test/models/User.spec.ts

```
@@ -321,6 +321,41 @@ describe("User", () => {
     });
   });
 
+  describe("delete profile", () => {
+    it("should delete an existing user", async () => {
+      let profile: any = {};
+
+      const caller = createCaller({
+        get: async (key: string) => {
+          switch (key) {
+            case "profile":
+              return JSON.stringify({
+                id: "id",
+                name: "Test",
+                email: "test@example.com",
+                tenantId: "tenantId",
+                created_at: "2021-01-01T00:00:00.000Z",
+                modified_at: "2021-01-01T00:00:00.000Z",
+                connections: [],
+              });
+          }
+        },
+        put: async (key: string, value: string) => {
+          switch (key) {
+            case "profile":
+              profile = JSON.parse(value);
+          }
+          return;
+        },
+        delete: async () => {},
+      });
+
+      await caller.delete();
+
+      // what is even happening here? looks like I'm testing a mock
```

All the above code is just guessing. Doesn't seem like I'm testing anything

---

So the get, put and delete are the mock for the storage. So what you need to validate is that the delete method on the storage is called.
If the get method returns null the delete function should throw a NotFound error as you're attempting to delete a non-existing user.